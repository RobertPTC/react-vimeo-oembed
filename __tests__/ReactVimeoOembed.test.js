import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ReactVimeoOembed } from '../index.js';

configure({ adapter: new Adapter() });

describe ('ReactVimeoOembed', () => {
  it ('should be defined', () => {
    expect(ReactVimeoOembed).toBeDefined();
  });

  it ('should render', () => {
    const tree = shallow(<ReactVimeoOembed />);
    expect(tree).toMatchSnapshot();
  });

  it('should call createEmbedScript when componentDidMount', () => {
    const spy = jest.spyOn(ReactVimeoOembed.prototype, 'createEmbedScript');
    const props = {
      videoId: 262524875,
      options: {}
    };
    const wrapper = mount(<ReactVimeoOembed {...props} />);
    expect(spy).toHaveBeenCalled();
    wrapper.unmount();
  });

  it('should have a script tag after mounting', () => {
    const props = {
      videoId: 262524875,
      options: {}
    };
    const wrapper = mount(<ReactVimeoOembed {...props} />);
    expect(wrapper.instance().script).toBeDefined();
    wrapper.unmount();
  });

  it('should have a callbackName', () => {
    global.embedVimeoVideoCallback3 = jest.fn();
    const props = {
      videoId: 262524875,
      options: {
        callback: () => null
      }
    };
    const wrapper = mount(<ReactVimeoOembed {...props} />);
    global.embedVimeoVideoCallback3({ html: '<div></div>' });
    expect(wrapper.instance().callbackName).toEqual('embedVimeoVideoCallback3');
    wrapper.unmount();
  });

  it('should change loading state to false when Vimeo data is fetched', () => {
    global.embedVimeoVideoCallback4 = jest.fn();
    const props = {
      videoId: 262524875,
      options: {
        callback: () => null
      }
    };
    const wrapper = mount(<ReactVimeoOembed {...props} />);
    global.embedVimeoVideoCallback4({ html: '<div></div>' });
    expect(wrapper.instance().state.videoLoading).toEqual(false);
    wrapper.unmount();
  });

  it('should set Vimeo embed player HTML in state when Vimeo data is fetched', () => {
    global.embedVimeoVideoCallback5 = jest.fn();
    const props = {
      videoId: 262524875,
      options: {
        callback: () => null
      }
    };
    const wrapper = mount(<ReactVimeoOembed {...props} />);
    global.embedVimeoVideoCallback5({ html: '<div></div>' });
    expect(wrapper.instance().state.videoEmbed).toEqual('<div></div>');
    wrapper.unmount();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
