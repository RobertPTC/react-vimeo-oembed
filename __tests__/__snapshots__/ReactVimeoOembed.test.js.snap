// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactVimeoOembed should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactVimeoOembed
    LoadingComponent={[Function]}
    className=""
    errorCallback={[Function]}
    options={
      Object {
        "api": false,
        "autopause": true,
        "autoplay": false,
        "byline": true,
        "color": "",
        "loop": false,
        "player_id": "",
        "portrait": true,
        "title": true,
        "xhtml": false,
      }
    }
    scriptLoadCallback={[Function]}
    style={Object {}}
    videoId=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
